tests/models/graph_tests.py

```diff
@@ -1067,3 +1068,25 @@ def test_appending_a_branch_with_an_invalid_ontology_class(self):
 
         with self.assertRaises(GraphValidationError) as cm:
             graph.save()
+
+    def test_appending_to_published_graph(self):
+        graph = Graph.objects.get(node=self.rootNode)
+        admin = User.objects.get(username="admin")
+        graph.publish(user=admin)
+        self.addCleanup(graph.unpublish)
+
+        with self.assertRaises(GraphValidationError) as cm:
+            graph.append_node()
+        self.assertEqual(cm.exception.code, 1012)
+        with self.assertRaises(GraphValidationError) as cm:
+            graph.append_branch("http://www.nasa.gov/", graphid=self.NODE_NODETYPE_GRAPHID)
+        self.assertEqual(cm.exception.code, 1012)
+
+    def test_appending_published_branch_to_unpublished_graph(self):
+        graph = Graph.objects.get(node=self.rootNode)
+        admin = User.objects.get(username="admin")
+        branch = Graph.objects.get(graphid=self.NODE_NODETYPE_GRAPHID)
+        branch.publish(user=admin)
+        self.addCleanup(branch.unpublish)
+
+        graph.append_branch("http://www.nasa.gov/", graphid=self.NODE_NODETYPE_GRAPHID)
```

Comment:
Happy to use a more realistic URL if wanted, perhaps by extracting a class constant that's easy to reuse.